### PR TITLE
added all changes and new features to documentation for Nov 2022

### PR DESCRIPTION
**Pull request by unserialize**
* changelog for 2 previous years
* job workers
* compile-time location
* function colors
* distcc and nocc
* mention FFI in various places
* generic functions
* generic classes
* all available @kphp- tags
* all available built-in functions
* all available cmd-line options for kphp2cpp
* all available cmd-line options for server/cli
* roadmap
* link to a Telegram chat

### CODE REVIEW

File: docs/kphp-language/howto-by-kphp/compile-time-location.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+big v=4 (in demo5 at dev.php:59)
+big v=5 (in demo5 at dev.php:59)
+end calculation (in demo5 at dev.php:61)
+```
+
+Notice, that `log_info()` prints out a parent (caller) location every time. It differs from time to time, it's unique for every call place. 
+
+Here's the implementation:
+```php
+function log_info(string $message, CompileTimeLocation $loc = null) {
```
Comment (Danil42Russia) [inline docs/kphp-language/howto-by-kphp/compile-time-location.md:32]:
```suggestion
function log_info(string $message, ?CompileTimeLocation $loc = null) {
```

File: docs/kphp-language/howto-by-kphp/colored-functions.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      'ssr ssr-allow-rpc rpc' => 1,
+    ]
+  ];
+}
+```
+
+
+## A demo project with colored functions
+
+Save [this folder from kphp-snippets](https://github.com/VKCOM/kphp-snippets/tree/master/colored-demo) somewhere. Execute
```
Comment (Danil42Russia) [inline docs/kphp-language/howto-by-kphp/colored-functions.md:51]:
```suggestion
Save [this folder from kphp-snippets]({{site.url_github_kphp_snippets}}/tree/master/colored-demo) somewhere. Execute
```

Comment (unserialize) [reply docs/kphp-language/howto-by-kphp/colored-functions.md:51]: +

**Comment by Ahmed-su-do** [comment]
في الأربعاء، ٢٦ أكتوبر ٢٠٢٢ ٥:٣٨ م Aleksandr Kirsanov <
***@***.***> كتب:

**Review by Danil42Russia — APPROVED** [review]

**Pull request merged**